mpas_analysis/ocean/time_series_antarctic_melt.py

```diff
@@ -0,0 +1,238 @@
+                                  self.regionMaskFileName))
+
+        # Load mesh related variables
+        try:
+            self.restartFileName = self.runStreams.readpath('restart')[0]
+        except ValueError:
+            raise IOError('No MPAS-O restart file found: need at least one '
+                          'restart file for Antarctic melt calculations')
+
+        # get a list of timeSeriesStats output files from the streams file,
+        # reading only those that are between the start and end dates
+        streamName = self.historyStreams.find_stream(
+            self.streamMap['timeSeriesStats'])
+        self.startDate = config.get('timeSeries', 'startDate')
+        self.endDate = config.get('timeSeries', 'endDate')
+        self.inputFiles = \
+            self.historyStreams.readpath(streamName,
+                                         startDate=self.startDate,
+                                         endDate=self.endDate,
+                                         calendar=self.calendar)
+
+        if len(self.inputFiles) == 0:
+            raise IOError('No files were found in stream {} between {} and '
+                          '{}.'.format(streamName, self.startDate,
+                                       self.endDate))
+
+        return  # }}}
+
+    def run(self):  # {{{
+        """
+        Performs analysis of the time-series output of Antarctic sub-ice-shelf
+        melt rates.
+
+        Authors
+        -------
+        Xylar Asay-Davis
+        """
+
+        print "\nPlotting Antarctic melt rate time series..."
+
+        print '  Load melt rate data...'
+
+        config = self.config
+        calendar = self.calendar
+
+        print '\n  Reading files:\n' \
+              '    {} through\n    {}'.format(
+                  os.path.basename(self.inputFiles[0]),
+                  os.path.basename(self.inputFiles[-1]))
+
+        # Load data:
+        variableList = ['timeMonthly_avg_landIceFreshwaterFlux']
+        timeVariableName = ['xtime_startMonthly', 'xtime_endMonthly']
+        ds = open_multifile_dataset(fileNames=self.inputFiles,
+                                    calendar=calendar,
+                                    config=config,
+                                    timeVariableName=timeVariableName,
+                                    variableList=variableList,
+                                    startDate=self.startDate,
+                                    endDate=self.endDate)
+
+        freshwaterFlux = ds.timeMonthly_avg_landIceFreshwaterFlux
+
+        movingAverageMonths = config.getint('timeSeriesAntarcticMelt',
+                                            'movingAverageMonths')
+
+        iceShelvesToPlot = config.getExpression('timeSeriesAntarcticMelt',
+                                                'iceShelvesToPlot')
+
+        dsRestart = xarray.open_dataset(self.restartFileName)
+        areaCell = dsRestart.areaCell
```

Comment:
The following line needs to be added here:
```python
landIceFraction = dsRestart.landIceFraction
```

Reply:
Ok, great! Well that was an easy fix (I didn't know landIceFraction even existed). I'll give this a try and report back.

Reply:
At first glance this looks much better. I'll do some more careful checks on the reported areas later this morning.

Reply:
With this change, the area of the indiv. shelves I checked are within ~5% and the total ice shelf area of the AIS is within ~7% (relative to the values reported in Rignot et al.). This seems reasonable to me considering the coarse res. of the 60to30 mesh. Also, all of the modeled shelf areas are biased too-small which also seems reasonable when you consider the fractal nature of coastlines.

Reply:
Great!  I don't recall the exact algorithm I used to get landIceFraction but there's definitely some smoothing involved.  I think there's also something that says, if a cell has more than 50% grounded ice, it's grounded, so that would likely account for the bias toward less ice-shelf area.  This should decrease with higher res but stay biased in the same direction.

Reply:
@xylar -- FYI, I am making code updates and pushing them to my fork now, so you don't have to make these same changes on your fork anymore if you don't want to.